nixos/modules/config/ipa.nix

```
@@ -0,0 +1,241 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.ipa;
+  sssd = config.services.sssd;
+  ntp = config.services.ntp;
+  krb5 = config.krb5;
+  ldap = config.users.ldap;
+  pyBool = x: if x then "True" else "False";
+
+  ldapConf = pkgs.writeText "ldap.conf" ''
+    # Turning this off breaks GSSAPI used with krb5 when rdns = false
+    SASL_NOCANON    on
+
+    URI ldaps://${cfg.server}
+    BASE ${cfg.basedn}
+    TLS_CACERT /etc/ipa/ca.crt
+  '';
+  nssDb = pkgs.runCommand "ipa-nssdb" { buildInputs = [ pkgs.nss.tools ]; } ''
+    set -x
+    mkdir -p $out
+    < /dev/urandom tr -dc '[:print:]' | head -c 40 > $out/pwdfile ||:
+    chmod 600 $out/pwdfile
+    certutil -d $out -N -f $out/pwdfile
+    chmod 644 $out/*.db
+    certutil -d $out -A -f $out/pwdfile -n "${cfg.realm} IPA CA" -t CT,C,C -i ${cfg.certificate}
+  '';
+
+in {
+  options = {
+    ipa = {
```

```suggestion
    services.ipa = {
```
This should be under some category.

---

It's not really a service though. It's a part of the machine definition. It defines that the machine connects to the freeipa server and uses it for logon etc.

---

We also have security or programs but it shouldn't really be outside of those categories.